src/Paket.Core/Versioning/Requirements.fs

```diff
@@ -791,25 +791,30 @@ type PackageRequirement =
     member this.Depth = this.Graph.Length
 
     static member Compare(x,y,startWithPackage:PackageFilter option,boostX,boostY) =
-        if x = y then 0 else
-        seq {
-            yield compare
-                (not x.VersionRequirement.Range.IsGlobalOverride,x.Depth)
-                (not y.VersionRequirement.Range.IsGlobalOverride,y.Depth)
-            yield match startWithPackage with
+        if obj.ReferenceEquals(x, y) then 0 else
+        let c = compare
+                  (not x.VersionRequirement.Range.IsGlobalOverride,x.Depth)
+                  (not y.VersionRequirement.Range.IsGlobalOverride,y.Depth)
+        if c <> 0 then c else
+        let c = match startWithPackage with
                     | Some filter when filter.Match x.Name -> -1
                     | Some filter when filter.Match y.Name -> 1
                     | _ -> 0
-            yield -compare x.ResolverStrategyForDirectDependencies y.ResolverStrategyForDirectDependencies
-            yield -compare x.ResolverStrategyForTransitives y.ResolverStrategyForTransitives
-            yield compare boostX boostY
-            yield -compare x.VersionRequirement y.VersionRequirement
-            yield compare x.Settings.FrameworkRestrictions y.Settings.FrameworkRestrictions
-            yield compare x.Parent y.Parent
-            yield compare x.Name y.Name
-        }
-        |> Seq.tryFind (fun x -> x <> 0)
-        |> Option.fold (fun _ x -> x) 0
+        if c <> 0 then c else
+        let c = -compare x.ResolverStrategyForDirectDependencies y.ResolverStrategyForDirectDependencies
+        if c <> 0 then c else
+        let c = -compare x.ResolverStrategyForTransitives y.ResolverStrategyForTransitives
+        if c <> 0 then c else
+        let c = compare boostX boostY
+        if c <> 0 then c else
+        let c = -compare x.VersionRequirement y.VersionRequirement
+        if c <> 0 then c else
+        let c = compare x.Settings.FrameworkRestrictions y.Settings.FrameworkRestrictions
+        if c <> 0 then c else
+        let c = compare x.Parent y.Parent
+        if c <> 0 then c else
+        let c = compare x.Name y.Name
+        if c <> 0 then c else 0
```

Comment:
Because the old code was a thing of beauty? :D

According to my benchmark (take it with a truckload of salt) it is slightly faster: https://gist.github.com/0x53A/b8bb15e040d152742fd067ce257e80a2

 Method |     Mean |     Error |    StdDev |  Gen 0 | Allocated |
------- |---------:|----------:|----------:|-------:|----------:|
    Old | 656.1 ns | 6.2845 ns | 4.9066 ns | 0.0508 |     264 B |
    New | 105.5 ns | 0.3383 ns | 0.2641 ns |      - |       0 B |